src/test/java/com/microsoft/sqlserver/jdbc/exception/ExceptionTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Microsoft JDBC Driver for SQL Server
+ * 
+ * Copyright(c) Microsoft Corporation All rights reserved.
+ * 
+ * This program is made available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+package com.microsoft.sqlserver.jdbc.exception;
+
+import static org.junit.Assert.fail;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.io.UnsupportedEncodingException;
+import java.net.SocketTimeoutException;
+import java.net.URI;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+
+import org.junit.jupiter.api.Test;
+import org.junit.platform.runner.JUnitPlatform;
+import org.junit.runner.RunWith;
+
+import com.microsoft.sqlserver.jdbc.SQLServerBulkCSVFileRecord;
+import com.microsoft.sqlserver.jdbc.SQLServerConnection;
+import com.microsoft.sqlserver.jdbc.SQLServerException;
+import com.microsoft.sqlserver.testframework.AbstractTest;
+
+@RunWith(JUnitPlatform.class)
+public class ExceptionTest extends AbstractTest {
+    static String inputFile = "BulkCopyCSVTestInput.csv";
+
+    /**
+     * Test the SQLServerException has the proper cause when encoding is not supported.
+     * 
+     * @throws Exception
+     */
+    @Test
+    public void testBulkCSVFileRecordExceptionCause() throws Exception {
+        String filePath = getCurrentClassPath();
+
+        try {
+            SQLServerBulkCSVFileRecord scvFileRecord = new SQLServerBulkCSVFileRecord(filePath + inputFile, "invalid_encoding", true);
+        }
+        catch (Exception e) {
+            if (!(e instanceof SQLServerException)) {
+                throw e;
+            }
+
+            assertTrue(null != e.getCause(), "Cause should not be null.");
+            assertTrue(e.getCause() instanceof UnsupportedEncodingException, "Cause should be instance of UnsupportedEncodingException.");
+        }
+    }
+
+    /**
+     * 
+     * @return location of resource file
+     */
+    static String getCurrentClassPath() {
```

Comment:
we have `getCurrentClassPath()` in BulkCopyCSVTest as well. I moved them into Utils class.